src/mlir/dialect/VeronaTypes.cc

```diff
@@ -106,6 +106,27 @@ namespace mlir::verona::detail
     }
   };
 
+  struct DescriptorTypeStorage : public TypeStorage
+  {
+    Type descriptor;
```

Comment:
Renamed to `StaticClassType`. I don't think we should bikeshed too much this name, but now it's very specific to what it means with enough comment on it to explain it further. This type may fuse with some other concepts in the future and we'll probably need to rename it, or if we just want to rename it as in it would be a simple commit.

Reply:
Sorry I didn't mean renaming the struct, but the field and getter. I actually liked `DescriptorType`.